### PR TITLE
chore: adjust owners file to the current state

### PR DESCRIPTION
**Pull request by mstruebing**
This brings the `OWNERS` file up to date.

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+- davidgamero
 reviewers:
 - brendandburns
-- drubin
-- itowlson 
+- mstruebing
+- davidgamero
 emeritus_approvers:
 - mbohlool # 10/22/2020
+- drubin # 11/23/2023
```
Comment (brendandburns) [inline OWNERS:12]: add itowlson here too.

**Comment by brendandburns** [comment]
One small comment to preserve @itowlson as an emeritus maintainer of this codebase, then lgtm.

File: OWNERS (hunk 1 above)
Comment (mstruebing) [reply OWNERS:12]: I've added him, I wasn't sure as the documentation explicitly states approvers but he was in the reviewers section.

**Pull request merged**